### ISSUE TITLE
Could not get a new token (login_required: Login required).

### ISSUE THREAD

**Issue by PeterC66** [auth0-samples/auth0-vue-samples#63]
I downloaded the sample app last week. There was a problem with it never requesting a logout for Auth0, which I reported on the [community site](https://community.auth0.com/t/sample-vue-spa-app-01-login-is-not-asking-for-userid-password/19088). The suggestion I was given seemed to work, but I then had a different issue - an alert saying `Could not get a new token (login_required: Login required)` which I could not get rid of.

Before reporting it here I upgraded to Node.js LTS (10), and did a new install of the sample app. I believe that I followed the given instructions. The same alert issue appears immediately, so it seems impossible to use. If it helps, my browser is Google Chrome (up to date - Version 71.0.3578.98 (Official Build) (64-bit)).

Is there something wrong with my set up?

I attach a screenshot.
![screenshot](https://user-images.githubusercontent.com/16805744/50218031-72e8c500-0382-11e9-880f-06df31a41224.jpg)

**Comment by albertoperdomo** [comment]
Hello @PeterC66,

thanks for taking the time to report this.
We are looking into this. Updates to follow.

Alberto.

**Comment by arlysea** [comment]
Hi Alberto...

I am also getting this popup message coming up straight away when I first run 'npm start' :

localhost:3000 says
Could not get a new token (error: redirectUri can't be empty).

Even though I have populated the AUTH_CONFIG.callbackUrl (auth0-variables.js) with a callback URL..

Regards
Andrew

**Comment by arlysea** [comment]
Hi Alberto...
Figured it out... not an issue with the repository... just some capitalisation mismatches...

Regards
Andrew

**Comment by albertoperdomo** [comment]
>just some capitalisation mismatches...

Hello @arlysea,
can you expand on this, please?

**Comment by arlysea** [comment]
Hi Alberto,
In my AuthService.js the param "redirectUri: AUTH_CONFIG.callbackURL," and in my auth0-variables.js the param was called "callbackUrl"... there was a mismatch between 'callbackURL' and 'callbackUrl'... (in capitalisation)

I was flicking between the different sample apps... '01-Login' and '05-Authorization' and I had saved my auth0-variables file, so I didn't have to get my clientID and domain details again... and there is a difference between the wording of 'callbackURL' which tripped me up...

Hope I am explaining clearly for you
Regards
Andrew

**Comment by albertoperdomo** [comment]
Thanks, Andrew.
This does not seem to be the issue we are seeing though. I just checked and the capitalization is ok on my side and the variable name matches.

**Comment by arlysea** [comment]
Yep... that's all good. 
My sample app is working fine now... I can log in and out with out any issues now.

Regards
Andrew

**Comment by PeterC66** [comment]
I have just tried my new version of your sample app afresh without changing anything. I once again got the alert saying `Could not get a new token (login_required: Login required)`. However this time when I clicked on OK it did not reappear. I was then able to Login and out with no problems.

**Except that** your current version of the sample app has the problem that it does not log out of Auth0 (per [this](https://community.auth0.com/t/sample-vue-spa-app-01-login-is-not-asking-for-userid-password/19088)). So once again I added `logoutUrl: 'http://localhost:3000/home',` to auth0-variables.js and in AuthService.js I replaced `router.replace('home')` by `this.auth0.logout({ returnTo: AUTH_CONFIG.logoutUrl })`. Then I got the ever-repeating alert saying `Could not get a new token (login_required: Login required)`. 

Maybe the issue is with this logout?

**Comment by PeterC66** [comment]
No.  I gave up on Auth0.  I am using a simple method based on [Jason Watmore's approach](http://jasonwatmore.com/post/2018/07/14/vue-vuex-user-registration-and-login-tutorial-example)

**Comment by Hardyng** [comment]
Probably the cause of problem is using Auth0's dev keys for social logins:

https://www.npmjs.com/package/auth0-js
In paragraph about checkSession method:

> Important: If you're not using the hosted login page to do social logins, you have to use your own social connection keys. If you use Auth0's dev keys, you'll always get login_required as an error when calling checkSession.

**Comment by elkdanger** [comment]
Closing this for now. If anyone continues to have this problem and has tried the suggestion from @Hardyng above, please feel free to file another issue.

**Comment by stphnlwlsh** [comment]
@stevehobbsdev Can Auth0 provide some _very_ clear documentation on this? I've spent the better part of two days troubleshooting why my `this.$auth.isAuthenticated` value disappears after a page refresh. Disabling all social logins fixes this, but this isn't the reason I chose Auth0. I generated some app client ids and secrets, and everything started lighting up again.

There are warning in the UI of Auth0 indicating that using the Auth0 keys isn't a good idea for production, but the message about the `checkSession` method is buried. Can that be elevated to be more prominent, somewhere for people in the future? There are no less than 10 forum topics with issues similar to this one.

Thanks!
